subsys/settings/src/settings.c

```diff
@@ -55,12 +56,19 @@ int settings_register(struct settings_handler *handler)
 			goto end;
 		}
 	}
+
+	handler->cprio = cprio;
 	sys_slist_append(&settings_handlers, &handler->node);
 
 end:
 	k_mutex_unlock(&settings_lock);
 	return rc;
 }
+
+int settings_register(struct settings_handler *handler)
+{
+	return settings_register_with_cprio(handler, 0);
```

Comment:
The default is 0 instead of INT_MIN, which will cause the load speed of the current default code to be slower when loading (although it is insignificant), but I think it would be better to change it to INT_MIN, because it will not affect the online code.

Reply:
You are right that this would result in (insignificant) lower commit speed. Setting the value to INT_MIN would make the default highest priority. I decided not to do that to allow users to be able to register handlers with faster than default/slower than default commit handlers. I think both cases should be available.

Reply:
I agree. I think it makes sense that the default is not the highest priority as that would prevent anyone from implementing anything with a higher priority than that.